sql/src/main/java/org/apache/druid/sql/calcite/schema/DruidSchema.java

```diff
@@ -402,6 +403,11 @@ void refresh(final Set<SegmentId> segmentsToRefresh, final Set<String> dataSourc
     // Rebuild the dataSources.
     for (String dataSource : dataSourcesToRebuild) {
       final DruidTable druidTable = buildDruidTable(dataSource);
+      if (druidTable == null) {
```

Comment:
Should we add a unit test around this that verifies the cleanup of a stale table?

Reply:
:+1: added a test, the added test fails with
```
java.lang.AssertionError: expected null, but was:<DruidTable{dataSource=wat, rowSignature={}}>
```
prior to the changes in this PR